core/foundation/inc/TClassEdit.h

```diff
@@ -153,6 +153,30 @@ namespace TClassEdit {
       TSplitType &operator=(const TSplitType &) = delete;
    };
 
+   /// A simple RAII helper to remove and readd enclosing _Atomic()
+   /// It expects no spaces at the beginning or end of the string
+   class AtomicTypeNameHandlerRAII {
+      std::string &fTypeName;
+      bool fIsAtomic = false;
+      const char *fgPrefix = "_Atomic(";
+
+   public:
+      AtomicTypeNameHandlerRAII(std::string &typeName) : fTypeName(typeName)
+      {
+         if (0 == fTypeName.find(fgPrefix)) {
```

Comment:
We should check and/or assert that `typeName.back() == ')'`

Reply:
excellent idea.